pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java

```diff
@@ -694,9 +694,11 @@ private static void convertSchema(SchemaInfo schemaInfo, Schema schema) {
                 .setType(getSchemaType(schemaInfo.getType()));
 
         schemaInfo.getProperties().entrySet().stream().forEach(entry -> {
-            schema.addProperty()
-                .setKey(entry.getKey())
-                .setValue(entry.getValue());
+            if (entry.getKey() != null && entry.getValue() != null) {
```

Comment:
Is it necessary to keep the key here to have a worthy situation? What do you think?

Reply:
The key must be non-null.

If you're using `PulsarAdmin` to create a schema whose properties have a null key, exception will be thrown that null key cannot be serialized to JSON. You can try to add a line `pulsarAdmin.schemas().createSchema(topicName, schemaInfo)` to `testNullKeyValueProperty` before creating the consumer, and see the output.

Therefore, I think from client's side, it should also skip the null key when using `Commands::convertSchema` to serialize a schema to `CommandProduce` or `CommandSubscribe`.

Reply:
ok